### DISCUSSION TITLE
Separate service to generate TTS and play it

### DISCUSSION THREAD

**Discussion by julianrinaldi** [nimroddolev/chime_tts#99] [Q&A] [answered]
It would be nice if there was a way to call a service to generate a TTS message before actually playing it. My specific use case is kind of ridiculous, but it would be nice to have. I have buttons for different timers in my gym, like 30 seconds, 60 seconds, 90 seconds, etc, that I press to have a rest period before different weight lifting exercises. Then at the end of the timer, I generate a TTS message through Eleven Labs of a voice of Rocky Balboa that generates different inspirational messages in his linguistic style using OpenAI. But both OpenAI and Eleven Labs add a combined delay of between 10 and 20 seconds. I have the timers set to do the TTS 15 seconds before the timer goes off. But it would be nice if it could be generated beforehand and then played at the exact moment the timer goes off. As far as I know this isn't currently possible.

**Comment by dbullendsl** [comment]
you can use the Chime TTS: Say URL  service to do that, it'll create the chime message and give you a URL for the output.

**Comment by nimroddolev** [comment]
In the next version of Chime TTS, the `chime_tts.say_url` service will include a `media_content_id` key value in its response which you will be able to use to call `media_player.play_media` at a later time, as you described.

For the time being @dbullendsl is correct in that you can use the `url` key value in the `chime_tts.say_url`'s response so that the notification is ready to be played at a later time.

**Reply by nimroddolev** [reply, marked as answer]
@julianrinaldi, with the release of v1.0.0 you can now prepare a TTS ahead of time, per your request.

To start using it, make sure to first configure your [`chime_tts.say_url` folder path](https://nimroddolev.github.io/chime_tts/docs/documentation/say_url-service/#media_content_id-key) to be inside your `/media` folder. You can then call the `chime_tts.say_url` service, which will return the `media_content_id` of the generated audio for you to use in your automations.

**Reply by julianrinaldi** [reply]
Awesome, thanks. I'll give it a try.
